GoogleUtilities/FileDataStorage/GULStorageHeartbeat.m

```diff
@@ -35,14 +35,46 @@ - (instancetype)initWithFileURL:(NSURL *)fileURL {
   return self;
 }
 
+- (nullable NSMutableDictionary *)getDictionary {
+  NSString *jsonString = [NSString stringWithContentsOfURL:self.fileURL
+                                                  encoding:NSUTF8StringEncoding
+                                                     error:nil];
+  NSError *jsonError;
+  if (jsonString == nil) {
+    return [NSMutableDictionary dictionary];
+  }
+  NSData *objectData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+  NSMutableDictionary *json = [NSJSONSerialization JSONObjectWithData:objectData
```

Comment:
[NSFileCoordinator](https://developer.apple.com/documentation/foundation/nsfilecoordinator?language=objc) looks like the best way to synchronize access to the file to me. It will cover both access from different threads of the same app as well as from different processes (in case of iOS app extensions). An example can be found [here](https://www.atomicbird.com/blog/sharing-with-app-extensions).

Reply:
as discussed offline this is not needed.